src/core/talent.js

```diff
@@ -196,18 +199,22 @@ class Talent extends IOFeatures {
         await this.pg.publishJson(ev.returnTopic, this.__createDiscoveryResponse());
     }
 
-    __getRules() {
-        const rules = this.getRules();
+    __getRules(triggerRules) {
+        triggerRules = triggerRules || this.getRules();
```

Comment:
It was not easy for me to understand what is happing here. I think some comments would help here that this is used and specify for the derived class FunctionTalent.

I propose to keep the talent.js class clean from any specific stuff which are specific for FunctionTalents. I assume this is possible (e.g. by overriding __createDiscoveryResponse in FunctionTalent to handle the triggerRules and combine them with the getRules & callees). 

The separation of getRules() and __getRules() is not specific and it enables us to override the internal handling for different derived classes.

Reply:
In my opinion, there's no specific implementation in the Talent class. The talent class still does not know, if there is a Function talent. It just offers additional trigger rules to be defined. This is either the ones coming from the another source (if trigger rules is given) or it can be the rules, which are implemented using getRules().

Reply:
For me it is specific, because it has no functionality in the base class itself and it also confused me a lot. (E.g. here this.getRules() is ignored if triggerRules and the derived class has to handle it the right way... )

Reply:
I'll check the newest  changes on that